espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/drivers/EspressoDriver.kt

```diff
@@ -0,0 +1,57 @@
+package io.appium.espressoserver.lib.drivers
+
+import android.view.View
+import androidx.test.espresso.PerformException
+import androidx.test.espresso.action.ViewActions
+import io.appium.espressoserver.lib.handlers.exceptions.InvalidArgumentException
+import io.appium.espressoserver.lib.handlers.exceptions.InvalidElementStateException
+import io.appium.espressoserver.lib.handlers.exceptions.InvalidSelectorException
+import io.appium.espressoserver.lib.handlers.exceptions.NoSuchElementException
+import io.appium.espressoserver.lib.helpers.ViewFinder
+import io.appium.espressoserver.lib.model.*
+import io.appium.espressoserver.lib.viewaction.ViewGetter
+import io.appium.espressoserver.lib.viewaction.ViewTextGetter
+
+class EspressoDriver : AppDriver {
+    override fun findElement(params: Locator): BaseElement {
+        var parentView: View? = null
+        params.elementId?.let {
+            parentView = ViewGetter().getView(Element.getViewInteractionById(it))
+        }
+        // Test the selector
+        val view = ViewFinder.findBy(
+            parentView,
+            params.using ?: throw InvalidSelectorException("Locator strategy cannot be empty"),
+            params.value ?: throw InvalidArgumentException()
+        )
+            ?: throw NoSuchElementException(
+                String.format(
+                    "Could not find element with strategy %s and selector %s",
```

Comment:
Is it helpful to address espresso context as the error message?

Reply:
added